lib/tasks/application.rake

```diff
@@ -121,4 +123,45 @@ namespace :application do
     Rails.logger = ActiveSupport::Logger.new($stdout)
     Rails.logger.level = 1
   end
+
+  namespace :links_valid do
+    desc "Checks the validity of links in division summary"
+    task divisions: :environment do
+      md = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+
+      Division.find_each do |division|
+        wiki_motion = division.wiki_motion
+
+        if wiki_motion
+          summary_in_html = md.render(wiki_motion.description)
+          parsed_data = Nokogiri::HTML.parse(summary_in_html)
+          url_hash_table = Hash.new(0)
+          broken_urls = []
+
+          tags = parsed_data.xpath("//a")
+          tags.each do |tag|
+            url_from_page = tag[:href]
+            url = url_from_page
+
+            # adds http to the start of the URL to avoid failure since web request will fail without http://
```

Comment:
This comment still doesn't actually explain why urls are appearing without "http://"

Reply:
Just in the case that a URL is missing the http://. As an edge case.

Reply:
Yes I understand but it still doesn't explain **why** that can happen. A url should always have "http://" or "https://" in front of it. The comment should explain why a non-valid url can appear in the markdown and if it isn't a valid url then why should it be hacked to be valid?

Reply:
I thought of it as an edge case. If you are saying that it should and will always have "http://" or "https://" then its better to be removed.

Reply:
It's the markdown renderer that is rendering the html with the links so it seems very unlikely to me that it would generate urls without "http(s)" if front of it and for those urls to work correctly in a browser but not be valid urls. So, yes, it makes sense to me to remove this. Maybe we'll find some other edge cases when we actually run this in production.